middleware/csrf/csrf.go

```diff
@@ -51,34 +51,12 @@ func New(config ...Config) fiber.Handler {
 
 		// Action depends on the HTTP method
 		switch c.Method() {
-		case fiber.MethodGet:
+		case fiber.MethodGet, fiber.MethodHead, fiber.MethodOptions, fiber.MethodTrace:
```

Comment:
RFC7231 defines GET HEAD OPTIONS and TRACE as safe, no need to validate the token, but check to see if the cookie is set, we generate token and set cookie later if not.